src/components/MewNotification/MewNotification.vue

```diff
@@ -1,29 +1,72 @@
 <template>
   <div
     @click="onToggle"
-    :class="[active ? 'activated' : '', notification.status.value + '-type', 'notification-container', 'px-3', 'titlePrimary--text']"
+    :class="[active ? 'activated' : '', notification.status.value + '-type', 'notification-container', 'px-3', 'titlePrimary--text', read ? 'read' : '']"
   >
     <v-container>
       <v-row>
         <v-col
           cols="6"
         >
           <div class="d-flex align-center">
-            <div :class="[ getClasses(notification.status.value.toLowerCase()), 'indicator', 'd-none', 'd-sm-flex']" />
+            <div
+              :class="[ getClasses(notification.status.value.toLowerCase()), 'indicator', 'd-none', 'd-sm-flex']"
+              v-if="!read"
+            />
             <mew-blockie
               class="d-none d-sm-flex"
-              v-if="notification.type.string.toLowerCase() !== txTypes.swap"
+              v-if="notification.type.value.toLowerCase() !== txTypes.swap"
               width="30px"
               height="30px"
-              address="0xDECAF9CD2367cdbb726E904cD6397eDFcAe6068D"
+              :address="notification.from.value"
             />
+            <div
+              v-else
+              class="d-none d-sm-flex flex-column currency-symbol"
+            >
+              <img
+                :src="notification.fromObj.icon"
+                width="30px"
+                height="30px"
+              >
+              <img
+                :src="notification.toObj.icon"
+                width="30px"
+                height="30px"
+                class="overlap"
+              >
+            </div>
             <div class="ml-5 detail-container full-width">
-              <div class="caption font-weight-medium d-flex">
+              <div
+                class="caption font-weight-medium d-flex"
+                v-if="notification.type.value.toLowerCase() !== txTypes.swap"
+              >
                 {{ notification.from.string }}: <span class="mew-address font-weight-medium ml-1 full-width"><mew-transform-hash :hash="notification.from.value" /> </span>
               </div>
-              <div class="caption font-weight-medium d-flex">
+              <div
+                class="caption font-weight-medium d-flex"
+                v-if="notification.type.value.toLowerCase() === txTypes.swap"
+              >
+                {{ notification.to.string }}: <span class="mew-address font-weight-medium ml-1 full-width"><mew-transform-hash :hash="notification.toObj.to" /> </span>
+              </div>
+              <div
+                class="caption font-weight-medium d-flex"
+                v-if="notification.type.value.toLowerCase() !== txTypes.swap"
+              >
                 {{ notification.amount.string }}: {{ notification.amount.value }}
               </div>
+              <div
+                class="caption mew-heading-2 d-flex"
+                v-if="notification.type.value.toLowerCase() === txTypes.swap"
```

Comment:
do you think itll be easier if this is computed property? other looks good!

Reply:
I was thinking that but I wasnt sure computed property was okay. Lemme move it there.